src/feature.js

```diff
@@ -149,29 +153,68 @@ class Feature {
         PROTECTION_DISABLED : PROTECTION_ENABLED;
     });
 
+    // Listen for the page to load to show the "is this page broken?"
+    // survey for the previous (reloaded) page.
+    browser.trackers.onPageDOMContentLoaded.addListener((tabId) => {
+      const tabInfo = TabRecords.getTabInfo(tabId);
+      if (tabInfo && tabInfo.payloadWaitingForSurvey) {
+        this.possiblyShowNotification(tabInfo);
+      }
+    });
+
+
     // Watch for the user pressing the "Yes this page is broken"
     // button and record the answer.
     browser.popupNotification.onReportPageBroken.addListener(
       (tabId) => {
+        const tabInfo = TabRecords.getOrInsertTabInfo(tabId);
+        if (!tabInfo || !tabInfo.payloadWaitingForSurvey) {
+          return;
+        }
+        tabInfo.payloadWaitingForSurvey.page_reloaded_survey = SURVEY_PAGE_BROKEN;
         this.recordSurveyInteraction(tabId, SURVEY_PAGE_BROKEN);
+        this.submitPayloadWaitingForSurvey(tabInfo);
       },
     );
 
     // Watch for the user pressing the "No this page is not broken"
     // button and record the answer.
     browser.popupNotification.onReportPageNotBroken.addListener(
       (tabId) => {
+        const tabInfo = TabRecords.getOrInsertTabInfo(tabId);
+        if (!tabInfo || !tabInfo.payloadWaitingForSurvey) {
+          return;
+        }
+        tabInfo.payloadWaitingForSurvey.page_reloaded_survey = SURVEY_PAGE_NOT_BROKEN;
         this.recordSurveyInteraction(tabId, SURVEY_PAGE_NOT_BROKEN);
+        this.submitPayloadWaitingForSurvey(tabInfo);
       },
     );
   }
 
+  submitPayloadWaitingForSurvey(tabInfo) {
```

Comment:
This seems to have been called twice when I only clicked "Other Reason" once. 😢 The first time had the expected values ...
```json
  "data": {
    "attributes": {
      "etld": "045c744075ff09a36c6072fe6cf0b7782d3b5797714a567be8d790b4dc9196b5",
      "num_blockable_trackers": "66",
      "page_reloaded": "true",
      "page_reloaded_survey": "3",
```

The next time had `-1` for `num_blockable_trackers` ?

```json
  "data": {
    "attributes": {
      "etld": "045c744075ff09a36c6072fe6cf0b7782d3b5797714a567be8d790b4dc9196b5",
      "num_blockable_trackers": "-1",
      "page_reloaded": "true",
      "page_reloaded_survey": "3",
```

Reply:
Note: I opened fox.com and reloaded once to get the survey, then answered once.

But the only other `shield-study-addon` ping I see in `about:telemetry` has this ...

```json
  "data": {
    "attributes": {
      "etld": "045c744075ff09a36c6072fe6cf0b7782d3b5797714a567be8d790b4dc9196b5",
      "num_blockable_trackers": "25",
      "page_reloaded": "true",
      "page_reloaded_survey": "0",
```

I don't have any ping with `page_reloaded_survey: 1` ? But I expect to have at least 1 ping with that to indicate that the survey was shown?

This "feels" like some kind of race condition between the new event emitters and handlers changing `tabInfo`?

Reply:
@groovecoder are you sure that this is sent twice? I ask because I can reproduce the `"num_blockable_trackers": "-1"` (not in FastBlock, though), but that ping is sent after I reload or navigate, as before. When are those two pings sent for you? The flow should go like this:

- visit fox.com -> document 1
- reload fox.com
- (note no telemetry is sent here)
- new (reloaded) document for fox.com opens -> document 2
- doorhanger is shown
- answer doorhanger
- telemetry ping for document 1 with `"page_reloaded_survey": "3",` is sent
- close tab (or other navigation action)
- telemetry ping for document 2 with `"page_reloaded_survey": "0",` is sent

> I don't have any ping with page_reloaded_survey: 1 ? But I expect to have at least 1 ping with that to indicate that the survey was shown?

`"page_reloaded_survey": "3",` implies that the survey was shown on that document (because the user answers it, so it's all good